### PR TITLE
Added "worker" to config resolution

### PR DESCRIPTION
**Pull request by christopheranderson**
We were missing "worker:" in the language config module.

### CODE REVIEW

**Comment by christopheranderson** [comment]
@mamaso - why didn't the Java tests fail? Isn't that code common to all workers?

**Comment by christopheranderson** [comment]
Ah, it looks like we're not doing a comparison for path there either, so we need tests for both?

**Comment by mamaso** [comment]
@christopheranderson the java tests are running only the `TryConfigureArguments` step. This bug was in the `WorkerConfigFactory` which is untested. We need to test that the `WorkerPath` is set correctly based on `DefaultWorkerPath` and `workers:<lang>:path` configuration.

**Comment by mamaso** [comment]
Essentially, test the GetConfigs function: https://github.com/Azure/azure-webjobs-sdk-script/pull/2043/files#diff-4a1578f8d75a14b4780790f365956ca4L24

File: test/WebJobs.Script.Tests/Description/Proxy/ProxyFunctionDescriptorProviderTests.cs [hunk 1]
```diff
@@ -42,6 +42,7 @@ public ProxyFunctionDescriptorProviderTests()
 
             _proxyClient = GetMockProxyClient();
             _settingsManager = ScriptSettingsManager.Instance;
+            _settingsManager.Reset();
```
Comment (mathewc) [inline test/WebJobs.Script.Tests/Description/Proxy/ProxyFunctionDescriptorProviderTests.cs]: ? Tests flaking out or something?

File: test/WebJobs.Script.Tests/Rpc/WorkerConfigFactoryTests.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                new TestWorkerProvider() {
+                    Language = lang, 
+                    Extension = extension, 
+                    DefaultWorkerPath = defaultWorkerPath
+                }
+            });
+
+            Assert.Single(workerConfigs);
+
+            foreach (var workerConfig in workerConfigs) {
```
Comment (mathewc) [inline test/WebJobs.Script.Tests/Rpc/WorkerConfigFactoryTests.cs]: above you're verifying Single, but here you're looping?

**Comment by mathewc** [comment]
Also, can you squash the 3 commits before checking in?

File: test/WebJobs.Script.Tests/Description/Proxy/ProxyFunctionDescriptorProviderTests.cs (hunk 1 above)
Comment (christopheranderson) [reply test/WebJobs.Script.Tests/Description/Proxy/ProxyFunctionDescriptorProviderTests.cs]: Yeah. Fabio recommended that as a fix. I guess we're planning on changing how it works later, anyway?

Comment (fabiocav) [reply test/WebJobs.Script.Tests/Description/Proxy/ProxyFunctionDescriptorProviderTests.cs]: Yes. This is temporary and caused by other tests essentially setting values that remain cached on the same instance. We'll be removing this type soon.

File: test/WebJobs.Script.Tests/Rpc/WorkerConfigFactoryTests.cs (hunk 2 above)
Comment (christopheranderson) [reply test/WebJobs.Script.Tests/Rpc/WorkerConfigFactoryTests.cs]: That is weird. I'll just do ElementAt(0).

**Comment by mamaso** [comment]
lgtm, squash and merge it in!

File: test/WebJobs.Script.Tests/Rpc/WorkerConfigFactoryTests.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            {
+                new TestWorkerProvider() {
+                    Language = lang, 
+                    Extension = extension, 
+                    DefaultWorkerPath = defaultWorkerPath
+                }
+            });
+
+            Assert.Single(workerConfigs);
+            Assert.Equal(workerConfigs.ElementAt(0).Arguments.WorkerPath, workerPath);
```
Comment (fabiocav) [inline test/WebJobs.Script.Tests/Rpc/WorkerConfigFactoryTests.cs]: nit: `workerConfigs.Single()` would essentially eliminate the need for the assert above this

File: test/WebJobs.Script.Tests/Rpc/WorkerConfigFactoryTests.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
+using System.Collections.Generic;
```
Comment (fabiocav) [inline test/WebJobs.Script.Tests/Rpc/WorkerConfigFactoryTests.cs]: Missing header

File: test/WebJobs.Script.Tests.Shared/TestWorkerProvider.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
+// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (fabiocav) [inline test/WebJobs.Script.Tests.Shared/TestWorkerProvider.cs]: Missing header

File: test/WebJobs.Script.Tests.Shared/TestWorkerProvider.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+{
+    public class TestWorkerProvider : IWorkerProvider
+    {
+        public string Language {get; set;}
+        public string Extension {get; set;}
+        public string DefaultWorkerPath {get; set;}
+
+        public WorkerDescription GetDescription() => new WorkerDescription
+        {
+            Language = this.Language,
```
Comment (fabiocav) [inline test/WebJobs.Script.Tests.Shared/TestWorkerProvider.cs]: nit: remove use of `this`

File: test/WebJobs.Script.Tests.Shared/TestWorkerProvider.cs [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests
+{
+    public class TestWorkerProvider : IWorkerProvider
+    {
+        public string Language {get; set;}
```
Comment (fabiocav) [inline test/WebJobs.Script.Tests.Shared/TestWorkerProvider.cs]: nit: spacing issue

**Comment by christopheranderson** [comment]
squashed and @fabiocav input is done

Can I get a :shipit:?

**Comment by fabiocav** [comment]
Looks like `WorkerConfigFactoryTests` is still missing a file header. Can you add that and :shipit: ?

**Comment by fabiocav** [comment]
LGTM. :shipit:

**Pull request merged**